### DISCUSSION TITLE
Image is cached and can't change it

### DISCUSSION THREAD

**Discussion by lighttigerXIV** [tauri-apps/tauri#6826] [Q&A]
Sorry if it feels like a dumb question but i can't find anything about it.

### Context
I am using React JS and tauri commands.
I am refactoring a app and decided to made it with tauri. It basically runs a image magick command and applies the catppuccin flavors LUTS to it. I managed to make that part work as intended, generate the LUTS and copy them to temp folder as latte.png or frappe.png and so on, and everything works fine at first.

### Problem
My problem is when i generate another image. The previous image with that flavor will be shown instead of the new one because it as the same path and name as the older one. I checked the image itself on the pc and it changes.
How can i replace the image with the new one? It feels like the image it's cached and i don't want that.

### What i tried
Basically i have this variables:
```javascript
const [noiseLevel, setNoiseLevel] = useState(4);
const [selectedImageRawPath, setSelectedImageRawPath] = useState("")
const [selectedImagePath, setSelectedImagePath] = useState("");
const [showGeneratedGrids, setShowGeneratedGrids] = useState(true);
const [generatedLattePath, setGeneratedLattePath] = useState("")
```

This is how i get the image:

```javascript
const selected = await open({
  multiple: false,
  filters: [{
    name: "Images",
    extensions: ["png", "webp", "jpg", "jpeg"]
  }]
});

if (selected !== null) {

  setSelectedImageRawPath(selected);
  setSelectedImagePath(convertFileSrc(selected));
  setGeneratedLattePath("")
}
```

This how i generate the images:
```javascript
invoke("generate_image", { image_path: selectedImageRawPath, noise_level: noiseLevel.toString(), flavor: 'latte' })
    .then((raw_path) => {

      setGeneratedLattePath(convertFileSrc(raw_path));
      console.log(raw_path)
      console.log(generatedLattePath)
    })
    .catch((error) => { console.error(error) })
```

And then set the image source as the path.

When searching i found a command called removed file from tauri fs and i was wondering if i should use that. I have tried with many paths but never got it right.

### Final thoughts

As you can see in the image, the file changed but the image in the sources didn't. How can i replace with the new one? 
If you have a solution in rust or javascript i would appreciate it. 
If you need more context i will reply as quickly as possible.

Thanks in advance :D

![image](https://user-images.githubusercontent.com/35658492/235425150-6f776c01-064e-456b-adf5-754900d53a76.png)

**Comment by FabianLars** [comment]
One thing i saw people doing (can't remember if it was with the asset protocol though) is to add random query parameters so that the img looks something similar to `<img src="image.png?12312313" />` and everytime you change the query param it should forcefully reload hopefully.

Edit: `image.png` of course needs to be replaced with the asset protocol/convertFileSrc fn

**Reply by Alii-isk** [reply]
thank u this worked for me
